lib/libthr/thread/thr_create.c

```diff
@@ -299,16 +301,19 @@ thread_start(struct pthread *curthread)
 	bool restore_sigmask;
 
 #ifdef CHERI_LIB_C18N
-	/*
-	 * At this point, curthread->tcb contains a fake wrapper TCB created by
-	 * RTLD when the thread was created. The real TCB has now been installed
-	 * by RTLD upon thread start, and the struct member should be set to it.
-	 */
-	curthread->tcb = _tcb_get();
-	restore_sigmask = true;
-#else
-	restore_sigmask = curthread->attr.suspend == THR_CREATE_SUSPENDED;
+	if (_rtld_c18n_is_enabled()) {
+		/*
+		 * At this point, curthread->tcb contains a fake wrapper TCB
+		 * created by RTLD when the thread was created. The real TCB has
+		 * now been installed by RTLD upon thread start, and the struct
+		 * member should be set to it.
+		 */
+		curthread->tcb = _tcb_get();
+		restore_sigmask = true;
```

Comment:
Am I missing something or is restore_sigmask always equal to create_suspended?

Reply:
That is my understanding too